### ISSUE TITLE
AdditionalFilesToRemoveFromTarget feature doesn't work when publishing from VS

### ISSUE THREAD

**Issue by rkeitel** [richardszalay/helix-publishing-pipeline#61]
This feature doesn't appear to remove any files from the web root. See [issue 52](https://github.com/richardszalay/helix-publishing-pipeline/issues/52).

I'm using version 1.5.4 of HPP. The publishing target is the local file system. The VS project uses package references. `DeleteExistingFiles` is set to `false`. `EnablePackageProcessLoggingAndAssert` is set to `true`, but the log file `AdditionalFilesToRemoveFromPublish.txt` is not created in `obj\Release\Package\Logs`.

**Comment by richardszalay** [comment]
Is your profile's publishUrl a file path or a web address?

**Comment by rkeitel** [comment]
The publishUrl is a file path.

**Comment by richardszalay** [comment]
Argh, you're killing me 🤣

I'll try to take a look, but if you could create a small repro and email it to me (richard@richardszalay.com) it will make diagnosing/fixing it quicker.

**Comment by rkeitel** [comment]
I can't create a repo because it is a customer project. But here you have the relevant configuration files:
[Testsite.zip](https://github.com/richardszalay/helix-publishing-pipeline/files/3391772/Testsite.zip)

**Comment by richardszalay** [comment]
@rkeitel Thanks for raising this, I can reproduce. For what it's worth, if you enable autopublish you won't hit this problem (which is why I've never seen it before)

I've logged a bug for this specific issue at #62, but you can use the following workaround at the end of your Local.pubxml:

```xml
<!-- Temporary workaround for https://github.com/richardszalay/helix-publishing-pipeline/issues/62 -->
<Target Name="_CollectAdditionalFilesToRemoveFromTarget"
        Condition="'$(BuildingInsideVisualStudio)' == 'true'"
        DependsOnTargets="CollectAdditionalFilesToRemoveFromTarget"
        BeforeTargets="_CopyWebApplication" />
```

**Comment by richardszalay** [comment]
Summary:

When expicitly publishing within VS the IDE's internal optimisations mean that a number of targets don't get run. Since appears to include WebFileSystemPublish, CollectAdditionalFilesToRemoveFromTarget never runs.

I think this hasn't been raised before because autopublish is not affected by it since it runs the targets by name. Still, we should create a new ticket that verifies each of the features when publishing from within VS.

**Comment by rkeitel** [comment]
Cool, the workaround works great. Thank you very much :-)

**Comment by richardszalay** [comment]
Excellent. I'll keep this issue open until I release a new version with the fix.

**Comment by rkeitel** [comment]
I noted another problem. Deleting additional assemblies is working (e.g. `<TargetPath>bin\Testsite.Feature.*</TargetPath>`), but removing additional configs is still not working (e.g. `<TargetPath>App_Config\Include\Feature\**</TargetPath>`).

**Comment by richardszalay** [comment]
I'm not sure if that glob is valid. Have you tried `App_Config\Include\Feature\**\*.config`?

**Comment by rkeitel** [comment]
Yes, I tried `App_Config\Include\Feature\*.config`, but it is not working.

**Comment by richardszalay** [comment]
App_Config\Include\Feature\*.config won't work unless the files aren't in any subfolders

App_Config\Include\Feature\** won't work because it will only match directories.

Can you try App_Config\Include\Feature\**\*.config ?

**Comment by rkeitel** [comment]
All files are in `App_Config\Include\Feature` without subfolders, so `App_Config\Include\Feature\*.config` should be the right one.

**Comment by richardszalay** [comment]
Interesting. Ok, let try to reproduce.

**Comment by richardszalay** [comment]
Even using your original config of `App_Config\Include\Feature\**` (+ the workaround I posted above), I can't actually reproduce this problem. For me, the extra config files go away when I publish via the VS dialog. I comment out the lines in Local.pubxml and reload the project (which is required, btw) then it stops doing it.

I've attached the sample project I used (which is just Helixbase with your config). Can you try it and see if you can reproduce it?

[working.zip](https://github.com/richardszalay/helix-publishing-pipeline/files/3401431/working.zip)

**Comment by richardszalay** [comment]
Ah good news. I'll close this issue once I've published a new version with the first fix.

**Comment by richardszalay** [comment]
Included in the [v1.5.5 release](https://www.nuget.org/packages/RichardSzalay.Helix.Publishing.WebRoot/1.5.5)
